TaskEngine/Tasks/TranscriptionTask.cs

```diff
@@ -21,39 +22,62 @@ class TranscriptionTask : RabbitMQTask<string>
         private readonly MSTranscriptionService _msTranscriptionService;
         private readonly GenerateVTTFileTask _generateVTTFileTask;
         private readonly SceneDetectionTask _sceneDetectionTask;
+        private readonly CaptionQueries _captionQueries;
+        private readonly CTDbContext _context;
 
         public TranscriptionTask(RabbitMQConnection rabbitMQ, MSTranscriptionService msTranscriptionService,
-            GenerateVTTFileTask generateVTTFileTask, SceneDetectionTask sceneDetectionTask, ILogger<TranscriptionTask> logger)
+            GenerateVTTFileTask generateVTTFileTask, SceneDetectionTask sceneDetectionTask, ILogger<TranscriptionTask> logger, CaptionQueries captionQueries, CTDbContext context)
             : base(rabbitMQ, TaskType.Transcribe, logger)
         {
             _msTranscriptionService = msTranscriptionService;
             _generateVTTFileTask = generateVTTFileTask;
             _sceneDetectionTask = sceneDetectionTask;
+            _captionQueries = captionQueries;
+            _context = context;
         }
+
+        /// <summary>
+        /// [2020.7.7] For the purpose of resuming failed transcriptions, all the captions of the failed transcription would now be stored in the database.
+        /// Before resuming, all the old captions would be queried out from the database and stored in a dictionary, which will be passed to the
+        /// transcription function along with the resume time point.
+        /// </summary>
+        /// <param name="videoId"></param>
+        /// <param name="taskParameters"></param>
+        /// <returns></returns>
         protected async override Task OnConsume(string videoId, TaskParameters taskParameters)
         {
             Video video;
-            using (var _context = CTDbContext.CreateDbContext())
+            video = await _context.Videos.Include(v => v.Video1).Where(v => v.Id == videoId).FirstAsync();
+            Key key = TaskEngineGlobals.KeyProvider.GetKey(video.Id);
+
+            // creat Dictionary and pass it to the recognition function
+            Dictionary<string, List<Caption>> captions = new Dictionary<string, List<Caption>>();
+            // check if query return empty List the first time
+            captions[Languages.ENGLISH] = await _captionQueries.GetCaptionsAsync(video.Id, Languages.ENGLISH);
```

Comment:
I thought I added a comment here, instead of calling ` await _captionQueries.GetCaptionsAsync` 5 times, add all the languages to a list, and iterate over the list

Reply:
Changed